service/src/main/java/bio/terra/workspace/common/stairway/StepUtils.java

```diff
@@ -0,0 +1,119 @@
+package bio.terra.workspace.common.stairway;
+
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+// Suppress sonar warnings about reflection API usage. Reflection APIs must be used to read and
+// write fields in a Step.
+@SuppressWarnings({"java:S3011"})
+public class StepUtils {
+
+  public static class MissingStepInputException extends RuntimeException {
+    public MissingStepInputException(String key) {
+      super("No flight value found for StepInput key '" + key + "'");
+    }
+  }
+
+  public static class IllegalSetException extends RuntimeException {
+    public IllegalSetException(Throwable cause) {
+      super(cause);
+    }
+  }
+
+  public static class IllegalGetException extends RuntimeException {
+    public IllegalGetException(Throwable cause) {
+      super(cause);
+    }
+  }
+
+  private StepUtils() {}
+
+  public static String keyFromField(Field field) {
```

Comment:
By default, the inputs and outputs are mapped to flight map (or flight input map) elements by using the annotated field's name. You can also provide an annotation argument to use a different name instead.